test/new-e2e/tests/agent-platform/persisting-integrations/persisting_integrations_test.go

```diff
@@ -0,0 +1,212 @@
+
+	vmOpts := []ec2.VMOption{}
+	if instanceType, ok := os.LookupEnv("E2E_OVERRIDE_INSTANCE_TYPE"); ok {
+		vmOpts = append(vmOpts, ec2.WithInstanceType(instanceType))
+	}
+
+	for _, osVers := range osVersions {
+		osVers := osVers
+		if platformJSON[*platform][*architecture][osVers] == "" {
+			// Fail if the image is not defined instead of silently running with default Ubuntu AMI
+			t.Fatalf("No image found for %s %s %s", *platform, *architecture, osVers)
+		}
+
+		t.Run(fmt.Sprintf("test upgrade persisting integrations on %s %s", osVers, *architecture), func(tt *testing.T) {
+			flake.Mark(tt)
+			tt.Parallel()
+			tt.Logf("Testing %s", osVers)
+
+			osDesc := platforms.BuildOSDescriptor(*platform, *architecture, osVers)
+			vmOpts = append(vmOpts, ec2.WithAMI(platformJSON[*platform][*architecture][osVers], osDesc, osDesc.Architecture))
+
+			e2e.Run(tt,
+				&persistingIntegrationsSuite{srcVersion: *srcAgentVersion},
+				e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(
+					awshost.WithEC2InstanceOptions(vmOpts...),
+				)),
+				e2e.WithStackName(fmt.Sprintf("upgrade-persisting-integrations-%s-test-%s-%v-%s", *srcAgentVersion, *flavorName, osVers, *architecture)),
+			)
+		})
+	}
+}
+
+func (is *persistingIntegrationsSuite) TestIntegrationPersistsWithFileFlag() {
+	VMclient := is.SetupTestClient()
+
+	startAgentVersion := is.SetupAgentStartVersion(VMclient)
+	is.InstallNVMLIntegration(VMclient)
+	is.PrepareMockedFiles(VMclient)
+
+	// set the flag to install third party deps
+	is.EnableInstallThirdPartyDepsFlag(VMclient)
+
+	upgradedAgentVersion := is.UpgradeAgentVersion(VMclient)
+	is.Require().NotEqual(startAgentVersion, upgradedAgentVersion)
+	is.CheckIntegrationInstalled(VMclient)
+}
+
+// func (is *persistingIntegrationsSuite) TestIntegrationDoesNotPersistWithoutFileFlag() {
+// 	VMclient := is.SetupTestClient()
+
+// 	startAgentVersion := is.SetupAgentStartVersion(VMclient)
+// 	is.InstallNVMLIntegration(VMclient)
+// 	is.PrepareMockedFiles(VMclient)
+
+// 	// unset the flag to install third party deps if it was set
+// 	VMclient.Host.MustExecute("sudo rm -f /opt/datadog-agent/.install_python_third_party_deps")
+
+// 	upgradedAgentVersion := is.UpgradeAgentVersion(VMclient)
+// 	is.Require().NotEqual(startAgentVersion, upgradedAgentVersion)
+// 	is.CheckIntegrationNotInstalled(VMclient)
+// }
+
+func (is *persistingIntegrationsSuite) SetupTestClient() *common.TestClient {
+	fileManager := filemanager.NewUnix(is.Env().RemoteHost)
+	agentClient, err := client.NewHostAgentClient(is, is.Env().RemoteHost.HostOutput, false)
+	require.NoError(is.T(), err)
+	unixHelper := helpers.NewUnix()
+	VMclient := common.NewTestClient(is.Env().RemoteHost, agentClient, fileManager, unixHelper)
+	return VMclient
+}
+
+func (is *persistingIntegrationsSuite) PrepareMockedFiles(VMclient *common.TestClient) {
+	for file, content := range filesToMock {
+		VMclient.Host.MustExecute(fmt.Sprintf("sudo bash -c \"echo '%s' > %s\"", content, file))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered using `host.WriteFile` instead? 

https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/pkg/utils/e2e/client/host.go#L250-L264

Reply:
Yes, I got a `Permission Denied` error, I had the same error with `host.CopyFile`

Reply:
It's probably executed without sudo, thank you for the context!